### PR TITLE
BDI and BGE: Do not include written off payments in data import matching

### LINKED ISSUE #4389: Make Written Off payments not count as open payments in BDI
Currently NPSP Batch Data Import will match to written off payments while matching donations.

In #4282 we've got the front end work done to exclude written off payments from donation matching - basically, not considering written off payments as open payments.

### PR DESCRIPTION
**Pull request by kenlewis-sfdo**
Prior to this change, Payments that were written-off were not excluded from the list of potential matches when the data import process (and the dry run data import process) ran.  This change excludes Payments that are written-off, eliminating the possibility of matching to a written-off Payment during these processes.

# Critical Changes

# Changes
- Batch Gift Entry will now exclude written-off Payments from the possible matches when you click Review Donations. Similarly, NPSP Data Import won't match donations to written-off payments.

# Issues Closed
Fixes #4389 

# New Metadata

# Deleted Metadata

### CODE REVIEW

File: src/classes/BDI_MatchDonations.cls [hunk 1]
```diff
@@ -468,30 +468,32 @@ public with sharing class BDI_MatchDonations implements BDI_IMatchDonations {
...
         // add Payment subquery
-        strSoql += ', (select ' + string.join(new list<String>(setPmtFields), ',');
-        strSoql += ' from npe01__OppPayment__r where npe01__Paid__c = false )';
-        
-        strSoql += ' from Opportunity';
-        strSoql += ' where IsClosed = false and (';
+        strSoql += ', (SELECT ' + string.join(new list<String>(setPmtFields), ',');
+        strSoql += ' FROM npe01__OppPayment__r ' +
+                'WHERE npe01__Paid__c = false ' +
+                'AND npe01__Written_Off__c = false)'; //Do not include written-off Payments.
```
Comment (ptewson-sfdo) [inline src/classes/BDI_MatchDonations.cls]: This line (AND npe01__Written_Off__c = false) is the only change not related to formatting and style, right?

**Comment by kenlewis-sfdo** [comment]
Yep, that's correct!

**Comment by kenlewis-sfdo** [comment]
**lurch: attach W-034100

**Comment by LurchTheButler** [comment]
Tracking <a href="https://foundation.my.salesforce.com/a2x1E000001C6ruQAC" target="_blank">W-034100</a>

**Pull request merged**